dev/conductor/core/test/repository_test.dart

```diff
@@ -501,6 +481,61 @@ enabled_branches:
         throwsExceptionWith('.ci.yaml already contains the branch master'),
       );
     });
+
+    test('repo set as localUpstream ensures requiredLocalBranches exist locally', () async {
```

Comment:
should we also check what happens when some of the branches do no exist? and also for EngineRepository?

Reply:
I could write a test; however, I would have to tell the FakeProcessManager exactly how to act when the git command was executed (that is throw an exception), and then I would be expecting that exception. So I would mainly be testing the FakeProcessManager, and not any of my code.

I will add an engine test.

Reply:
done, PTAL